src/edu/agh/wfiis/designpatterns/patternsUsageExample/patterns/SimpleDesignPatternCreator.java

```diff
@@ -0,0 +1,42 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.patterns;
+import edu.agh.wfiis.designpatterns.patternsUsageExample.enums.DesignPatternType;
+import edu.agh.wfiis.designpatterns.patternsUsageExample.enums.Pattern;
+
+public class SimpleDesignPatternCreator extends DesignPatternCreator
+{
+
+    public static String listAvailablePatterns()
+    {
+        return Pattern.values().toString();
+    }
+
+    @Override public DesignPattern create(Pattern pattern)
+    {
+        switch (pattern)
```

Comment:
Zamiast używać switcha, można zastanowic się nad wykorzystaniem polimorfizmu i stworzeniem trzech osobnych klas dla buildera, dekoratora i strategii.Ułatwi to późniejsze rozwijanie programu w przypadku dodania kolejnego wzorca. 
https://stackoverflow.com/questions/7149449/open-closed-principle-how-to-deal-with-this-switch